### DISCUSSION TITLE
Disable the `no-multi-assign` rule

### DISCUSSION THREAD

**Discussion by leroykorterink** [mediamonks/eslint-config#222] [General]
There is a proposal to disable the `no-multi-assign` rule.

> `no-multi-assign`: again, single liners can keep the code cleaner and faster to read.
> \- @edgardz

**Comment by leroykorterink** [comment]
Chaining the assignment of variables can lead to unexpected results and be difficult to read.

```ts
(function() {
    const foo = bar = 0; // Did you mean `foo = bar == 0`?
    bar = 1;             // This will not fail since `bar` is not constant.
})();

console.log(bar);
// 1
```

Please see [the documentation](https://eslint.org/docs/latest/rules/no-multi-assign) of the rule.

**Reply by ThaNarie** [reply]
Curious to see some examples  from project code where multi-assign was actually preferable.

I can't imagine that assigning the same `number`, `string` or `boolean` would happen that often ? Potentially only when resetting values to their empty state (`0`, `''`, `false`, `null`, `undefined`)?

And then, how often those cases would occur (and thus could be solved by a disable-line in those cases).
